azerothcore/azerothcore-wotlk
# refactor(DB/SQL): Update integer & utf to MySQL's new standards

## Changes Proposed:
MySQL gonna do some changes in future updates and throws these warnings when you make tables

- Remove specification of integer
- change utf8 & utf8mb3 to utf8mb4  
- float unsigned to CHECK (`col`>=0) (https://github.com/azerothcore/azerothcore-wotlk/issues/4930)
![image](https://user-images.githubusercontent.com/24550914/111892461-01ce6700-89fc-11eb-850f-24561f95ac57.png)

- Change varchar(500) to TEXT(500) in map_dbc due to 
```
Row size too large. The maximum row size for the used table type, not counting BLOBs, is 65535. This includes storage overhead, check the manual. You have to change some columns to TEXT or BLOBs
```

- Change version.sql engine from MyISAM to INNODB due to
```
ERROR 1071 (42000) at line 2776530: Specified key was too long; max key length is 1000 bytes
```
## Tests Performed:
- None



## How to Test the Changes:

re-make DB and see that everything works as expected

Guessing you haven't done witchcraft with your DB, here you can check all CHECK constraints
```sql
SELECT * FROM information_schema.CHECK_CONSTRAINTS;
```

**Note to ppl with HeidiSQL:** Check constraints works, but viewing them will be fixed with version 11.3 when it releases.


## Need to be tested with

- [x] MariaDB
- [x] MySQL 8.0
- [x] MySQL 5.7

## Target Branch(es):








 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

Note to self RegEx \(.+?\)
